src/components/grid_overlay/grid_overlay.ts

```diff
@@ -59,17 +59,23 @@ function useCellHovered(
     }
   }
 
+  function shouldUpdateMousePosition(e: MouseEvent) {
+    if (gridRef.el === e.target) {
+      return updateMousePosition(e);
+    }
```

Comment:
Put the condition inside `updateMousePosition`. 
1) it's 2 lines
2) we expect a function named "should..." to return a boolean, not to update the position if needed